Change hardcoded references to /usr/local and /usr/X11R6 to $(PREFIX). This
has been reported to the developer by email but they have not reacted.

Fix the install target not to use sudo, not to use the GNU-specific -t flag,
and to use DESTDIR. This has been reported to the developer by email but
they have not reacted.

Disable warning-to-error conversion for implicit function declarations
because otherwise it won't compile with Xcode 12 or later. The separate
ambfor makefile already does this. Doing this makes the port incompatible
with the arm64 architecture. This project has a zillion implicit function
declarations which it is too much work for me to fix myself; the developer
should do it. This has been reported to the developer by email but they have
not reacted.

Remove duplicate declarations of CFLAGS, FFLAGS, LIBS, and LIBSG. The first
FFLAGS has the vital -fallow-argument-mismatch flag that lets this software
compile with gcc10 and later. The second FFLAGS appends some unnecessary
duplicate flags so remove it.

Remove -fallow-argument-mismatch from CFLAGS because it is not a flag a C
compiler understands. Only a Fortran compiler understands this and the flag
is already in (the first copy of) FFLAGS.
--- docker/makefile.orig	2022-07-01 04:42:29.000000000 -0500
+++ docker/makefile	2023-03-03 02:22:29.000000000 -0600
@@ -96,7 +96,7 @@
 
 EXT= -std=gnu89 -Wno-return-type -DDARWIN ${EXTRAZ}
 FFLAGS=-O3 -funroll-loops -fallow-argument-mismatch
-CFLAGS= -g -std=gnu89 -Wno-return-type -DDARWIN -Wno-logical-op-parentheses -Wno-tautological-pointer-compare -Wno-tautological-constant-out-of-range-compare -fallow-argument-mismatch
+CFLAGS= -g -std=gnu89 -Wno-return-type -DDARWIN -Wno-logical-op-parentheses -Wno-tautological-pointer-compare -Wno-tautological-constant-out-of-range-compare -Wno-implicit-function-declaration
 LDR = ${FC}
 endif
 
--- makefile.orig	2022-12-08 07:09:06.000000000 -0600
+++ makefile	2023-03-03 00:59:43.000000000 -0600
@@ -16,7 +16,7 @@
 CC = cc
 LIBS =  -lX11 -lm
 LDR = ${FC} 
-LIBSG = -L/usr/X11R6/lib -lGLU -lGL -lXmu -lX11 -lm
+LIBSG = -L$(PREFIX)/lib -lGLU -lGL -lXmu -lX11 -lm
 ARCH := $(shell getconf LONG_BIT)
 AFLAG= -m$(ARCH)
 EXTEN=
@@ -132,18 +132,11 @@
 # Mac OS X g77
 #
 CC=cc
-CFLAGS= -std=gnu89 -Wno-return-type -DDARWIN -I/usr/X11R6/include -DDOBACK -DHASTIMER -DCOLOR_OFFSET=0.0 -Wno-logical-op-parentheses
-#FFLAGS=-O3 -funroll-loops -fallow-argument-mismatch
 FFLAGS+=-O3 -funroll-loops -fallow-argument-mismatch
-
-LIBS = -L/usr/X11R6/lib -lX11 -lm
-LIBSG = -L/usr/X11R6/lib -lGLU "-Wl,-dylib_file,/System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib:/System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib" -lGL -lXmu -lX11 -lm
 EXT= -std=gnu89 -Wno-return-type -DDARWIN ${EXTRAZ}
-#FFLAGS=-O3 -funroll-loops
-FFLAGS+=-O3 -funroll-loops
-LIBS = -L/usr/X11R6/lib -lX11 -lm
-LIBSG = -L/usr/X11R6/lib -lGLU "-Wl,-dylib_file,/System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib:/System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib" -lGL -lXmu -lX11 -lm
-CFLAGS= -g -std=gnu89 -Wno-return-type -DDARWIN -I/usr/X11R6/include -DDOBACK -DHASTIMER -DCOLOR_OFFSET=0.0 -Wno-logical-op-parentheses -Wno-tautological-pointer-compare -Wno-tautological-constant-out-of-range-compare -fallow-argument-mismatch
+LIBS = -L$(PREFIX)/lib -lX11 -lm
+LIBSG = -L$(PREFIX)/lib -lGLU "-Wl,-dylib_file,/System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib:/System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib" -lGL -lXmu -lX11 -lm
+CFLAGS= -g -std=gnu89 -Wno-return-type -DDARWIN -I$(PREFIX)/include -DDOBACK -DHASTIMER -DCOLOR_OFFSET=0.0 -Wno-logical-op-parentheses -Wno-tautological-pointer-compare -Wno-tautological-constant-out-of-range-compare -Wno-implicit-function-declaration
 LDR = ${FC}
 endif
 
@@ -274,7 +267,7 @@
 # and the full opengl version of molden 'gmolden', (does not need glut)
 # (make gmolden)
 #
-LIBSOGL = -lglut -lGLU -lGL -lXmu -lX11 -lm
+LIBSOGL = -L$(PREFIX)/lib -lglut -lGLU -lGL -lXmu -lX11 -lm
 #
 # on linux :
 #
@@ -371,7 +364,7 @@
 endif
 
 exten2:
-	utils/register_extension.sh /usr/local/bin
+	utils/register_extension.sh $(PREFIX)/bin
 
 install:	$(EXTENZ)
-	sudo install -t /usr/local/bin -m 755 bin/molden bin/gmolden bin/ambfor bin/ambmd bin/surf
+	install -m 755 bin/molden bin/gmolden bin/ambfor bin/ambmd bin/surf $(DESTDIR)$(PREFIX)/bin
--- src/surf/Makefile.orig	2022-05-06 15:51:50.000000000 -0500
+++ src/surf/Makefile	2022-05-14 03:23:26.000000000 -0500
@@ -6,7 +6,7 @@
 LINCLUDE    = -lm
 OPT_CFLAGS  = -O2 $(FLAGS) $(INCLUDE)
 #CFLAGS	    = -g $(FLAGS) $(INCLUDE)
-CFLAGS	    = -Wno-implicit-int -O2 $(FLAGS) $(INCLUDE) ${EXT}
+CFLAGS	    = -Wno-implicit-int -O2 $(FLAGS) $(INCLUDE) ${EXT} -Wno-implicit-function-declaration
 
 ifeq ($(uname), Darwin)
 #
